tiledb/libtiledb.pyx

```diff
@@ -2584,7 +2584,6 @@ cdef class Dim(object):
                 check_error(ctx,
                     tiledb_dimension_set_filter_list(ctx.ptr, dim_ptr, filter_list_ptr))
         except:
-            tiledb_dimension_free(&dim_ptr)
```

Comment:
Does this get freed somewhere else now?

Reply:
It is freed in `tiledb_dimension_alloc` on error. Initially when I implemented this, I was checking whether `dim_ptr` was null before freeing, but that doesn't work b/c of https://github.com/TileDB-Inc/TileDB/pull/1959. So we don't need to free at all here.

Reply:
see https://github.com/TileDB-Inc/TileDB/blob/97d400117232fbaf396398ce02637541f99ceb05/tiledb/sm/c_api/tiledb.cc#L1793-L1817 (before my commint in 1959)